cartridges/plugin_gtm/cartridge/scripts/gtm/gtmHelpers.js

```diff
@@ -276,50 +432,124 @@ function getConfirmationData(res, step) {
 
 /**
  * @param {object} res - current route response object
+ * @return {Object} Object containing confirmation page data.
+ */
+function getGA4ConfirmationData(res) {
+    var order = null;
+
+    try {
+        if ('token' in request.httpParameterMap) {
+            order = dw.order.OrderMgr.getOrder(res.order.orderNumber, request.httpParameterMap.token.value);
+        } else {
+            order = dw.order.OrderMgr.getOrder(res.order.orderNumber);
+        }
+    } catch (e) {
+        Logger.error('GTMHelpers - cannot retrieve order');
+    }
+
+    if (order) {
+        var obj = {
+            'event': 'purchase',
+            'ecommerce': {
+                'currency': order.currencyCode,
+                'transaction_id': order.orderNo,
+                'value': order.getAdjustedMerchandizeTotalPrice(true).getValue().toFixed(2),
+                'shipping': order.getAdjustedShippingTotalPrice().getValue().toFixed(2),
+                'tax': order.getTotalTax().getValue().toFixed(2),
+                'items': module.exports.getProductArrayFromList(order.getProductLineItems().iterator(), module.exports.getGA4OrderProductObject, true),
+                'affiliation': Site.getCurrent().getID()
+            }
+        };
+
+        var coupons = getCoupons(order.getCouponLineItems().iterator());
+        if (!empty(coupons)) {
+            obj.ecommerce['coupon'] = coupons;
+        }
+
+        return obj;
+    }
+
+    return {};
+}
+
+/**
+ * @param {Object} res - current route response object
+ * @param {Boolean} ga4 - is for GA4
  * @returns {Object} Object containing full datalayer
  */
-function getDataLayer(res) {
-    switch (res.action) {
-        case SITE_NAME:
-        case 'Home-Show':
-        case 'Default-Start':
-            return module.exports.getHomeData();
-        case 'Product-Show':
-        case 'Product-ShowInCategory':
-            return module.exports.getPdpData(res);
-        case 'Search-Show':
-            return module.exports.getSearchImpressionData(res);
-        case 'Cart-Show':
-            return module.exports.getCheckoutData(1);
-        case 'Checkout-Login':
-            return module.exports.getCheckoutData(2);
-        case 'Checkout-Begin':
-            return module.exports.getCheckoutData(3);
-        case 'CheckoutShippingServices-SubmitShipping':
-            return module.exports.getCheckoutData(4);
-        case 'CheckoutServices-SubmitPayment':
-            return module.exports.getCheckoutData(5);
-        case 'CheckoutServices-PlaceOrder':
-            return module.exports.getCheckoutData(6);
-        case 'Order-Confirm':
-            return module.exports.getConfirmationData(res, 7);
-    };
+function getDataLayer(res, ga4) {
+    if (ga4) {
+        // GA4 Events
+        switch (res.action) {
+            case 'Product-Show':
```

Comment:
Missing the `Product-ShowInCategory` case, this is a newer route in SFRA 5.x, but it should be there in both GA4 and Universal Analytics.